Add installation check scripts

kan-bayashi opened the pull request:
Added installation check scripts and how-to-run in README.

kan-bayashi commented:
Running example is as follows:
```
$ source venv/bin/activate && python check_install.py
INFO: python version = 2.7.5 (default, Aug  4 2017, 00:39:18)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
INFO: chainer is installed.
INFO: cupy is installed.
INFO: inflect is installed.
INFO: librosa is installed.
INFO: matplotlib is installed.
INFO: soundfile is installed.
INFO: torch is installed.
INFO: unidecode is installed.
INFO: pytorch warpctc is installed.
INFO: chainer warpctc is installed
INFO: torch version is matched.
INFO: chainer version is matched.
INFO: cuda is available in torch.
INFO: cudnn is available in torch.
INFO: cuda is available in chainer.
INFO: cudnn is available in chainer.
INFO: nccl is installed.
INFO: multi-gpu is available (#gpus = 10).
```

kan-bayashi commented:
@jnishi Is it possible to add some information to your forked warpctc?
If you can, it is helpful to check the problem is related to the warpctc version.

sw005320 commented on tools/check_install.py:
```diff
@@ -0,0 +1,116 @@
+logging.info("python version = " + sys.version)
+
+# check library availableness
+try:
```
Is it possible to get the some of packages from `tools/requirements.txt` including the version information (e.g., librosa, chainer with version 4.3.1)? Then, if someone add something in tools/requirements.txt, we may not have to change this script. (but it would be probably difficult).

kan-bayashi commented:
I updated to get the library list from `requirements.txt`.
Also we can add libraries to `MANUALLY_INSTALLED_LIBRARIES` in `check_install.py`.

```
INFO: python version = 2.7.5 (default, Aug  4 2017, 00:39:18)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
INFO: library availableness check start.
INFO: # libraries to be checked = 12
INFO: --> torch is installed.
INFO: --> torch version is matched.
INFO: --> chainer is installed.
INFO: --> chainer version is matched.
INFO: --> cupy is installed.
INFO: --> python_speech_features is installed.
INFO: --> setuptools is installed.
INFO: --> librosa is installed.
INFO: --> soundfile is installed.
INFO: --> inflect is installed.
INFO: --> unidecode is installed.
INFO: --> matplotlib is installed.
INFO: --> chainer_ctc is installed.
INFO: --> warpctc_pytorch is installed.
INFO: library availableness check done.
INFO: 12 / 12 libraries are correctly installed.
INFO: cuda availableness check start.
INFO: --> cuda is available in torch.
INFO: --> cudnn is available in torch.
INFO: --> cuda is available in chainer.
INFO: --> cudnn is available in chainer.
INFO: --> nccl is installed.
INFO: --> multi-gpu is available (#gpus = 10).
INFO: cuda availableness check done.
```

kan-bayashi commented:
I cleaned up `Makefile` and `conda.mk`.
- updated to be able to specify python
- separated patch processing
- modified conda.mk to install conda pytorch

jnishi commented:
@kan-bayashi Sure. But what kind of information would be helpful when original repository is not versioned.

kan-bayashi commented:
@jnishi Now we have the version check script.
If you add `warpctc_pytorch.__version__`, we can check the version is same.
https://github.com/espnet/espnet/blob/master/tools/check_install.py#L12
https://github.com/espnet/espnet/blob/master/tools/check_install.py#L65
I just want to check the user uses your patched version.
How about `warpctc.__version__ = 0.1.1`? (because original = 0.1 via `pip list`)

jnishi commented:
@kan-bayashi I added version info and updated to `0.1.1` please check. https://github.com/jnishi/warp-ctc/pull/4
